### PR TITLE
[Test] DO NOT MERGE gtest

### PR DESCRIPTION
**Pull request by yinghai**
test workaround for ninja with cmake `ExternalProject_Add`

@zrphercule

### CODE REVIEW

File: cmake/external/googletest.cmake [hunk 1]
```diff
@@ -1,30 +1,31 @@
...
+  set(googletest_STATIC_LIBRARIES
+      ${CMAKE_CURRENT_BINARY_DIR}/googletest/src/googletest/googletest/libgtest.a)
+endif()
+
+ExternalProject_Add(googletest
+    PREFIX googletest
+    GIT_REPOSITORY ${googletest_URL}
+    GIT_TAG ${googletest_TAG}
+    DOWNLOAD_DIR "${DOWNLOAD_LOCATION}"
+    BUILD_BYPRODUCTS ${googletest_STATIC_LIBRARIES}
```
Comment (yinghai) [inline cmake/external/googletest.cmake]: One line change.

File: cmake/external/googletest.cmake [hunk 2]
```diff
@@ -1,30 +1,31 @@
...
-    BUILD_IN_SOURCE 1
-    BUILD_COMMAND ${CMAKE_COMMAND} --build . --config Release --target gtest
-    INSTALL_COMMAND ""
-    CMAKE_CACHE_ARGS
-        -DCMAKE_BUILD_TYPE:STRING=Release
-        -DBUILD_GMOCK:BOOL=OFF
-        -DBUILD_GTEST:BOOL=ON
-        -Dgtest_force_shared_crt:BOOL=OFF
-)
+include (ExternalProject)
```
Comment (Maratyszcza) [inline cmake/external/googletest.cmake]: Why does every line show up as changed?

Comment (yinghai) [reply cmake/external/googletest.cmake]: Changed file from Dos to Linux format.

**Comment by yinghai** [comment]
Looks like it flushed out new issues.

**Comment by zrphercule** [comment]
Seems like we dont need to merge this pr anymore, since we are having other solutions but not trun on the ONNX_BUILD_TEST flag =)

**Comment by yinghai** [comment]
Test done.

**Pull request closed without merging**